src/Model/Item.php

```diff
@@ -1160,26 +1158,28 @@ public static function getIdAndNickByGuid($guid, $uid = 0)
 
 		// Does the given user have this item?
 		if ($uid) {
-			$r = q("SELECT `item`.`id`, `user`.`nickname` FROM `item` INNER JOIN `user` ON `user`.`uid` = `item`.`uid`
-				WHERE `item`.`visible` = 1 AND `item`.`deleted` = 0 AND `item`.`moderated` = 0
-					AND `item`.`guid` = '%s' AND `item`.`uid` = %d", dbesc($guid), intval($uid));
-			if (DBM::is_result($r)) {
-				$id = $r[0]["id"];
-				$nick = $r[0]["nickname"];
+			$items = dba::p("SELECT `item`.`id`, `user`.`nickname` FROM `item`
+				INNER JOIN `user` ON `user`.`uid` = `item`.`uid`
+				WHERE `item`.`visible` AND NOT `item`.`deleted` AND NOT `item`.`moderated`
+					AND `item`.`guid` = ? AND `item`.`uid` = ?", $guid, $uid);
+			if ($item = dba::fetch($items)) {
+				$id = $item["id"];
+				$nick = $item["nickname"];
 			}
 		}
 
 		// Or is it anywhere on the server?
 		if ($nick == "") {
-			$r = q("SELECT `item`.`id`, `user`.`nickname` FROM `item` INNER JOIN `user` ON `user`.`uid` = `item`.`uid`
-				WHERE `item`.`visible` = 1 AND `item`.`deleted` = 0 AND `item`.`moderated` = 0
+			$items = dba::p("SELECT `item`.`id`, `user`.`nickname` FROM `item`
```

Comment:
You can use `dba::fetch_first` for that instead.

Reply:
Same as above. We currenty can't work with joins in this function.

Reply:
Same as above.